File: backend/app/schemas/local_codex.py
# SPDX-FileCopyrightText: 2025 Weibo, Inc.
#
# SPDX-License-Identifier: Apache-2.0

"""Schemas for local Codex thread discovery and binding."""

from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class LocalCodexThreadSummary(BaseModel):
    """Summary of a local Codex thread discovered on a device."""

    model_config = ConfigDict(populate_by_name=True)

    thread_id: str = Field(..., alias="threadId")
    title: str
    cwd: Optional[str] = None
    updated_at: Optional[str] = Field(default=None, alias="updatedAt")
    archived: bool = False
    running: bool = False


class LocalCodexBindRequest(BaseModel):
    """Request to bind a local Codex thread to a Wework task."""

    model_config = ConfigDict(populate_by_name=True)

    device_id: str = Field(..., alias="deviceId", min_length=1)
    thread_id: str = Field(..., alias="threadId", min_length=1)
    team_id: int = Field(..., alias="teamId")
    title: Optional[str] = None
    cwd: Optional[str] = None


class LocalCodexBindResponse(BaseModel):
    """Response after binding or reusing a local Codex thread task."""

    model_config = ConfigDict(populate_by_name=True)

    task_id: int = Field(..., alias="taskId")
    task: dict
    created: bool
    thread_id: str = Field(..., alias="threadId")
    device_id: str = Field(..., alias="deviceId")


class LocalCodexThreadListResponse(BaseModel):
    """Response for local Codex thread discovery."""

    threads: list[LocalCodexThreadSummary]

File: wework/src/api/localCodex.ts
import type {
  LocalCodexBindRequest,
  LocalCodexBindResponse,
  LocalCodexThreadListResponse,
  LocalCodexThreadSummary,
} from '@/types/api'
import type { HttpClient } from './http'

export function createLocalCodexApi(client: HttpClient) {
  return {
    async listLocalCodexThreads(
      deviceId: string,
      limit = 50,
    ): Promise<LocalCodexThreadSummary[]> {
      const query = new URLSearchParams()
      query.set('limit', String(limit))
      const response = await client.get<LocalCodexThreadListResponse>(
        `/local-codex/devices/${encodeURIComponent(deviceId)}/threads?${query.toString()}`,
      )
      return response.threads
    },
    bindLocalCodexThread(
      request: LocalCodexBindRequest,
    ): Promise<LocalCodexBindResponse> {
      return client.post<LocalCodexBindResponse>('/local-codex/threads/bind', request)
    },
  }
}
